tutorials/models_mapping/plot_1_tensor_models.py

```diff
@@ -0,0 +1,333 @@
+
+    - Surface topography
+    - Adding structures of various shape to the model
+    - Parameterized models
+    - Models with 2 or more physical properties
+    
+
+"""
+
+#########################################################################
+# Import modules
+# --------------
+#
+
+from SimPEG import Mesh
+from SimPEG import Utils
+from SimPEG import Maps
+import numpy as np
+import matplotlib.pyplot as plt
+
+# sphinx_gallery_thumbnail_number = 3
+
+#############################################
+# Defining the mesh
+# -----------------
+#
+# Here, we create the tensor mesh that will be used for all examples.
+#
+
+
+def makeExampleMesh():
+
+    dh = 5.
+    hx = [(dh, 5, -1.3), (dh, 20), (dh, 5, 1.3)]
+    hy = [(dh, 5, -1.3), (dh, 20), (dh, 5, 1.3)]
+    hz = [(dh, 5, -1.3), (dh, 20), (dh, 5, 1.3)]
+    mesh = Mesh.TensorMesh([hx, hy, hz], 'CCC')
+
+    return mesh
+
+
+#############################################
+# Halfspace model with topography at z = 0
+# ----------------------------------------
+#
+# In this example we generate a half-space model. Since air cells remain
+# constant during geophysical inversion, the number of model values we define
+# should be equal to the number of cells lying below the surface. Here, we
+# define the model (*mod* ) as well as the mapping (*modMap* ) that goes from
+# the model-space to the entire mesh.
+#
+
+mesh = makeExampleMesh()
+
+m_val = 100.
+
+# Find cells below topography and define mapping
+m_air = 0.
+actv = mesh.gridCC[:, 2] < 0.
+modMap = Maps.InjectActiveCells(mesh, actv, m_air)
+
+# Define the model
+mod = m_val*np.ones(actv.sum())
+
+# We can plot a slice of the model at Y=-2.5
+Fig = plt.figure(figsize=(5, 5))
+Ax = Fig.add_subplot(111)
+mesh.plotSlice(modMap*mod, normal='Y', ax=Ax, ind=int(mesh.nCy/2), grid=True)
+Ax.set_title('Model slice at y = -2.5 m')
+plt.show()
+
+#############################################
+# Topography, a block and a vertical dyke
+# ---------------------------------------
+#
+# In this example we create a model containing a block and a vertical dyke
+# that strikes along the y direction. The utility *surface2ind_topo* is used
+# to find the cells which lie below a set of xyz points defining a surface.
+#
+
+mesh = makeExampleMesh()
+
+m_back = 100.
+m_dyke = 40.
+m_block = 70.
+
+# Define surface topography as an (N, 3) np.array. You could also load a file
+# containing the xyz points
+[xx, yy] = np.meshgrid(mesh.vectorNx, mesh.vectorNy)
+zz = -3*np.exp((xx**2 + yy**2) / 75**2) + 40.
+topo = np.c_[Utils.mkvc(xx), Utils.mkvc(yy), Utils.mkvc(zz)]
+
+# Find cells below topography and define mapping
+m_air = 0.
+actv = Utils.surface2ind_topo(mesh, topo, 'N')
+modMap = Maps.InjectActiveCells(mesh, actv, m_air)
+
+# Define the model on subsurface cells
+mod = m_back*np.ones(actv.sum())
+k_dyke = (mesh.gridCC[actv, 0] > 20.) & (mesh.gridCC[actv, 0] < 40.)
+mod[k_dyke] = m_dyke
+k_block = ((mesh.gridCC[actv, 0] > -40.) & (mesh.gridCC[actv, 0] < -10.) &
```

Comment:
sorry, this is me being picky but for long statements, would you mind opening the bracket and then entering to a new line (helps with readability and consistency - the tabs are then consistent along the entire program rather than dancing around). 
```          
k_block = (
    (mesh.gridCC[actv, 0] > -40.) & (mesh.gridCC[actv, 0] < -10.) &
    (mesh.gridCC[actv, 1] > -30.) & (mesh.gridCC[actv, 1] < 30.) &
    (mesh.gridCC[actv, 2] > -40.) & (mesh.gridCC[actv, 2] < 0.)
)
```

Reply:
fixed these according to pep8